iceoryx_utils/include/iceoryx_utils/log/logger.inl

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IOX_UTILS_LOG_LOGGER_INL
+#define IOX_UTILS_LOG_LOGGER_INL
+namespace iox
+{
+namespace log
+{
+inline bool Logger::IsEnabled(const LogLevel logLevel) const noexcept
```

Comment:
why the `logger.inl`? There is already a `logger.cpp`.

Reply:
Previously this check was done inside public API Logger::Log()
Now Logger::Log() calls IsEnabled(). 
Hence this is marked as inline , not to loose the performance as compared to  reference.

Reply:
@prasannabhat could you please explain why it has to be an inline method?

Reply:
The inline keyword has no use here, as it is not templated and thus can be implemented in a cpp file. As there should not be multiple definitions if the implementation is moved to a cpp file it can just be removed.

Reply:
> @prasannabhat could you please explain why it has to be an inline method?

Previously check against set LogLevel was done inside public API Logger::Log() (which is called for every logging operation)
Logger::Log() delegates this to IsEnabled().
Hence IsEnabled() is marked as inline , not to loose the performance compared to previous revision.